app/routes/goal_routes.py

```diff
@@ -0,0 +1,82 @@
+from app.models.task import Task
+from app import db
+from app.models.goal import Goal
+import os
+import datetime
+
+# from ..helpers import validate_goal
+from ..helpers import validate_object
+# from ..helpers import validate_task
+from flask import Blueprint, request, jsonify, make_response, abort
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_json()["goal"] for goal in goals]
+    return jsonify(goals_response), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def handle_goal(goal_id):
+    goal = validate_object(Goal, goal_id)
+    return jsonify(goal.to_json()), 200
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def handle_goal_task(goal_id):
+    goal = validate_object(Goal, goal_id)
+    response = goal.to_json_task()
+    for item in goal.tasks:  # for every task in goals
+        response["tasks"].append(item.to_json()["task"])
+    return jsonify(response), 200
+
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.create(request_body)
+        db.session.add(new_goal)
+        db.session.commit()
+
+    except KeyError:
+        return abort(make_response(jsonify({"details": "Invalid data"}), 400))
+
+    return new_goal.to_json(), 201
```

Comment:
to stay consistent with our response return statement, let's use `jsonify` on this like the other routes

```suggestion
    return jsonify(new_goal.to_json()), 201
```